groups/reconcile.go

```diff
@@ -0,0 +1,392 @@
+	g2, err := srv.Groups.Get(groupEmailId).Do()
+	if err != nil {
+		switch err := err.(type) {
+		case *googleapi.Error:
+			if err.Code == http.StatusNotFound {
+				fmt.Printf("skipping updating group settings as group %s has not yet been created\n", groupEmailId)
+				return
+			} else {
+				log.Fatalf("Unable to retrieve group info for group %s: %v", groupEmailId, err)
+			}
+		default:
+			log.Fatalf("Unable to retrieve group info for group %s: %v", groupEmailId, err)
+		}
+	}
+	if g2.AllowExternalMembers != "true" {
+		if !config.DryRun {
+			_, err := srv.Groups.Patch(groupEmailId, &groupssettings.Groups{
+				AllowExternalMembers: "true",
+			}).Do()
+			if err != nil {
+				log.Fatalf("Unable to update group info for group %s: %v", groupEmailId, err)
+			} else {
+				fmt.Printf("> Successfully updated group settings for %s to allow external members\n", groupEmailId)
+			}
+		} else {
+			fmt.Printf("dry-run : skipping updating group settings\n")
+		}
+	}
+}
+
+func addMembersToGroup(service *admin.Service, groupEmailId string, members []string) {
+	l, err := service.Members.List(groupEmailId).Do()
+	if err != nil {
+		switch err := err.(type) {
+		case *googleapi.Error:
+			if err.Code == http.StatusNotFound {
+				fmt.Printf("skipping adding members to group %s as it has not yet been created\n", groupEmailId)
+				return
+			} else {
+				log.Fatalf("Unable to retrieve members in group %s: %v", groupEmailId, err)
+			}
+		default:
+			log.Fatalf("Unable to retrieve members in group %s: %v", groupEmailId, err)
+		}
+	}
+
+	for _, m := range members {
+		found := false
+		if len(l.Members) > 0 {
+			for _, m2 := range l.Members {
+				if m2.Email == m {
+					found = true
+					break
+				}
+			}
+		}
+		if !found {
+			if !config.DryRun {
+				_, err := service.Members.Insert(groupEmailId, &admin.Member{
+					Email: m,
+					Role:  "MEMBER",
+				}).Do()
+				if err != nil {
+					log.Fatalf("Unable to add %s to %s : %v", m, groupEmailId, err)
+				} else {
+					fmt.Printf("Added %s to %s as a MEMBER\n", m, groupEmailId)
+				}
+			} else {
+				fmt.Printf("dry-run : Skipping adding %s to %s\n", m, groupEmailId)
+			}
+		}
+	}
+}
+
+func removeMembersFromGroup(service *admin.Service, groupEmailId string, members []string) {
+	l, err := service.Members.List(groupEmailId).Do()
+	if err != nil {
+		switch err := err.(type) {
+		case *googleapi.Error:
+			if err.Code == http.StatusNotFound {
+				fmt.Printf("skipping removing members group %s as group has not yet been created\n", groupEmailId)
+				return
+			} else {
+				log.Fatalf("Unable to retrieve members in group %s: %v", groupEmailId, err)
+			}
+		default:
+			log.Fatalf("Unable to retrieve members in group %s: %v", groupEmailId, err)
+		}
+	}
+
+	for _, m := range l.Members {
+		found := false
+		if len(members) > 0 {
+			for _, m2 := range members {
+				if m2 == m.Email {
+					found = true
+					break
+				}
+			}
+		}
+		if !found {
```

Comment:
Can you flip these negative booleans to just say `if found ...` and `if config.DryRun`?